packages/core/src/rpc/index.ts

```diff
@@ -116,15 +176,25 @@ export const createRequestQueue = ({
     initialStart: true,
     browser: false,
     worker: async (task: {
-      request: EIP1193Parameters<PublicRpcSchema>;
+      request: EIP1193Parameters<PublicRpcSchema> | SubscribeParameters;
       stopClockLag: () => number;
     }) => {
       common.metrics.ponder_rpc_request_lag.observe(
         { method: task.request.method, network: network.name },
         task.stopClockLag(),
       );
 
-      return await fetchRequest(task.request);
+      if (task.request.method === "eth_subscribe") {
+        return await withRetry({
+          fn: subscribe,
+          request: task.request,
+        });
```

Comment:
What are the difference between the errors throw in `subscribe()` vs the errors passed to `onError()` in the subscribe params

Reply:
I was confused a bit too, as far as I know, the onError handles the errors thrown during the active websocket connection, while the errors thrown in subscribe happen when establishing this connection. Feel free to correct me here.

Reply:
I'm not sure either but that makes sense